src/firefly/java/edu/caltech/ipac/firefly/server/servlets/AnyFileUpload.java

```diff
@@ -53,95 +57,128 @@ protected void processRequest(HttpServletRequest req, HttpServletResponse res) t
     }
 
     public static void doFileUpload(HttpServletRequest req, HttpServletResponse res) throws Exception {
+        UploadFileInfo fi = null;
+        File uf;
+        File destDir = null;
+        String ext;
+        String fileName = null;
+        String rPathInfo = "";
+        String fileType;
+        String fileAnalysis;
+        String overrideCacheKey = null;
+        FileType fType = null;
+        FileInfo urlDownloadInfo = null;
+        FileItemStream file = null;
 
         if (! ServletFileUpload.isMultipartContent(req)) {
-            sendReturnMsg(res, 400, "Is not a Multipart request. Request rejected.", "");
-        }
-        StopWatch.getInstance().start("Upload File");
+            SrvParam sp = new SrvParam(req.getParameterMap());
+            String wsCmd = sp.getOptional("wsCmd");
 
-        ServletFileUpload upload = new ServletFileUpload();
-        FileItemIterator iter = upload.getItemIterator(req);
-        FileItemStream file = null;
-        String url = null;
+            if (wsCmd == null) {
+                sendReturnMsg(res, 400, "Is not a Multipart request. Request rejected.", "");
+            }
```

Comment:
This was really confusing and it took me a while to understand.
This servlet was only handling multi-part upload and now it's changed to handle files from workspace as well.
I would parse all of the parameters first, then move the logic down to determine which kind of upload is requested.